### PR TITLE
Trb 17

### PR DESCRIPTION
**Pull request by bleaksmile**

### CODE REVIEW

File: src/main/java/com/greenfox/tribes1/Building/BuildigService.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.greenfox.tribes1.Building;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BuildigService {
```
Comment (ChrisFrill) [inline src/main/java/com/greenfox/tribes1/Building/BuildigService.java]: BuildigService --> please refactor to BuildingService

Comment (ChrisFrill) [inline src/main/java/com/greenfox/tribes1/Building/BuildigService.java]: BuildigService --> please refactor to BuildingService

**Review by ChrisFrill — CHANGES_REQUESTED** [review]

**Review by ChrisFrill — APPROVED** [review]

File: src/main/java/com/greenfox/tribes1/Building/Barracks.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.greenfox.tribes1.Building;
+
+public class Barracks extends Building {
+}
```
Comment (lyancsie) [inline src/main/java/com/greenfox/tribes1/Building/Barracks.java]: The singular form is "Barrack"

File: src/main/java/com/greenfox/tribes1/Building/BuildingService.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  @Autowired
+  public BuildingService(BuildingRepository buildingRepository) {
+    this.buildingRepository = buildingRepository;
+  }
+
+  public boolean isValidBuilding(Building building){
+    return building!=null;
+  }
+
+  public Building saveBuilding(Building building){
```
Comment (csadesz) [inline src/main/java/com/greenfox/tribes1/Building/BuildingService.java]: Missing spaces after methods, and at the if statement.
Should look like:

if () {
}

File: src/main/java/com/greenfox/tribes1/Building/BuildingType.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package com.greenfox.tribes1.Building;
+
+public enum BuildingType {
```
Comment (csadesz) [inline src/main/java/com/greenfox/tribes1/Building/BuildingType.java:3]: Nice solution :)

File: src/test/java/com/greenfox/tribes1/Building/BuildingServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+public class BuildingServiceTest {
+
+  BuildingService buildingService;
+  @Mock
+  BuildingRepository buildingRepository;
+
+  Building barrack = new Barracks();
```
Comment (csadesz) [inline src/test/java/com/greenfox/tribes1/Building/BuildingServiceTest.java]: These could be private

**Review by csadesz — CHANGES_REQUESTED** [review]

**Review by csadesz — APPROVED** [review]

**Pull request merged**